### PR TITLE
SDK Updates for Scheduler Jobs

### PR DESCRIPTION
**Pull request by alickrxu**
* Adds `pythAdapter` to constructor
* Add various getter methods needed for scheduler

### CODE REVIEW

**Review by consvic — APPROVED** [review]
LGTM

**Review by consvic — CHANGES_REQUESTED** [review]
Actually can you update the sdk script please 🙏🏼

**Comment by consvic** [comment]
This is the file https://github.com/cega-fi/cega-sdk-evm/blob/main/src/testScripts/sdkV2-scripts.ts
you only need to add the new address to the constructor

File: src/sdkV2.ts [hunk 1]
```diff
@@ -490,4 +522,14 @@ export default class CegaEvmSDKV2 {
     const oracleEntry = await this.loadOracleEntry();
     return oracleEntry.getPrice(baseAsset, quoteAsset, timestamp, oracleDataSource);
   }
+
+  async pythUpdateAssetPrices(
+    timestamp: number,
```
Comment (vsai) [inline src/sdkV2.ts:527]: this should be a `Date`, and in the function you can convert it into the corresponding `number`

the SDK *caller* shouldn't have to be concerned about 1701052220 vs 1701032220. but rather `"Monday 2pm UTC" etc.

File: src/sdkV2.ts [hunk 2]
```diff
@@ -490,4 +522,14 @@ export default class CegaEvmSDKV2 {
...
+    assetAddresses: EvmAddress[],
+    updates: string,
+    fee: number,
```
Comment (vsai) [inline src/sdkV2.ts:530]: fee should be a `ethers.BigNumberish`

if you're submitting a value, it can be a bignumber

File: src/sdkV2.ts [hunk 3]
```diff
@@ -490,4 +522,14 @@ export default class CegaEvmSDKV2 {
...
+  ): Promise<ethers.providers.TransactionResponse> {
+    const pythAdapter = await this.loadPythAdapter();
+    return pythAdapter.updateAssetPrices(timestamp, assetAddresses, updates, { value: fee });
```
Comment (vsai) [inline src/sdkV2.ts:533]:
```
function updateAssetPrices(
        uint40 timestamp,
        address[] calldata assets,
        bytes[] calldata updateDatas
```

if addresses takes in an array, then `updates` should also be taking in an array

File: src/sdkV2.ts [hunk 4]
```diff
@@ -21,9 +22,12 @@ export default class CegaEvmSDKV2 {
 
   private _treasuryAddress: EvmAddress;
 
+  private _pythAdapterAddress: EvmAddress;
+
   constructor(
     addressManager: EvmAddress,
     treasuryAddress: EvmAddress,
+    pythAdapterAddress: EvmAddress,
```
Comment (vsai) [inline src/sdkV2.ts:30]: `pythAdapterAddress` will almost never be used by:
- cega-client
- 3rd party sdk users

To make it simpler for most callers, can you make this optional, and put it after the signer so we don't have to be concerned about not including it when we instantiate the SDK

**Pull request merged**